Tasks/GooglePlayReleaseV4/GooglePlay.ts

```diff
@@ -0,0 +1,489 @@
+
+        return filesList;
+    }
+
+    return [];
+}
+
+function getVersionCodeListInput(): number[] {
+    const versionCodeFilterInput: string[] = tl.getDelimitedInput('replaceList', ',', false);
+    const versionCodeFilter: number[] = [];
+    const incorrectCodes: string[] = [];
+
+    for (const versionCode of versionCodeFilterInput) {
+        const versionCodeNumber: number = parseInt(versionCode.trim(), 10);
+
+        if (versionCodeNumber && (versionCodeNumber > 0)) {
+            versionCodeFilter.push(versionCodeNumber);
+        } else {
+            incorrectCodes.push(versionCode.trim());
+        }
+    }
+
+    if (incorrectCodes.length > 0) {
+        throw new Error(tl.loc('IncorrectVersionCodeFilter', JSON.stringify(incorrectCodes)));
+    } else {
+        return versionCodeFilter;
+    }
+}
+
+/**
+ * If any of the provided inputs are set, it will show a warning
+ * @param inputs inputs to check
+ */
+function warnIfUnusedInputsSet(...inputs: string[]): void {
+    for (const input of inputs) {
+        tl.debug(`Checking if unused input ${input} is set...`);
+        const inputValue: string | undefined = tl.getInput(input);
+        if (inputValue !== undefined && inputValue.length !== 0) {
+            tl.warning(tl.loc('SetUnusedInput', input));
+        }
+    }
+}
+
+/**
+ * Get obb file. Returns any file with .obb extension if present in parent directory else returns
+ * from apk directory with pattern: main.<versionCode>.<packageName>.obb
+ * @param apkPath apk file path
+ * @param packageName package name of the apk
+ * @param versionCode version code of the apk
+ * @returns ObbPathFile of the obb file is present else null
+ */
+function getObbFile(apkPath: string, packageName: string, versionCode: number): string | null {
+    const currentDirectory: string = path.dirname(apkPath);
+    const parentDirectory: string = path.dirname(currentDirectory);
+
+    const fileNamesInParentDirectory: string[] = fs.readdirSync(parentDirectory);
+    const obbPathFileInParent: string | undefined = fileNamesInParentDirectory.find(file => path.extname(file) === '.obb');
+
+    if (obbPathFileInParent) {
+        tl.debug(`Found Obb file for upload in parent directory: ${obbPathFileInParent}`);
+        return path.join(parentDirectory, obbPathFileInParent);
+    }
+
+    const fileNamesInApkDirectory: string[] = fs.readdirSync(currentDirectory);
+    const expectedMainObbFile: string = `main.${versionCode}.${packageName}.obb`;
+    const obbPathFileInCurrent: string | undefined = fileNamesInApkDirectory.find(file => file.toString() === expectedMainObbFile);
+
+    if (obbPathFileInCurrent) {
+        tl.debug(`Found Obb file for upload in current directory: ${obbPathFileInCurrent}`);
+        return path.join(currentDirectory, obbPathFileInCurrent);
+    } else {
+        tl.debug(`No Obb found for ${apkPath}, skipping upload`);
+        return null;
+    }
+}
```

Comment:
Could you please extract all secondary functions into separate modules? Let's try to keep only the main function in the main exec file.

Reply:
Done in https://github.com/microsoft/google-play-vsts-extension/pull/292/commits/33eef3925da4f5a3ce386de1964beee789d364c1
Need to decide what to do with `prepareTrackUpdate` function